src/thor/bidirectional_astar.cc

```diff
@@ -962,6 +964,9 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
 
     // Update edgelabel_index and transition cost to apply at next iteration
     edgelabel_index = predidx;
+    // We apply the turn cost at the beginning of the edge, as is done in the forward path
+    // Semantically this can be thought of is, how much time did it take to turn onto this edge
+    // To do this we need to carry the cost forward to the next edge in the path so we cache it here
```

Comment:
important for the reverse path algorithms is that on the reverse path you need to take the transition cost from the previous edge and put it at the beginning of the next edge.